src/systems/filecoin_nodes/repository/ipldstore/_index.md

```diff
@@ -1,29 +1,35 @@
+
+<!-- TODO: insert final version of https://github.com/ipld/specs/issues/244#issuecomment-637269338 when ready, it'll land on the IPLD specs repo README -->
+
+At its core, IPLD defines a [Data Model](https://github.com/ipld/specs/blob/master/data-model-layer/data-model.md) for representing data. The Data Model is designed for practical implementation across a wide variety of programming languages, while being as useful as possible for content-addressed data and a broad range of generalized tools to interact with that data. The Data Model includes a range of standard primitive types (or "kinds"), such as booleans, integers, strings, nulls and byte arrays, as well as two recursive types: lists and maps. Because IPLD is designed for content-addressed data, it also includes a "link" primitive in its Data Model. In practice, links use the [CID](https://github.com/multiformats/cid) specification. IPLD data is organized into "blocks", where a block is represented by the raw, encoded data and its content-address, or CID. Every content-addressable chunk of data can be represented as a block, and together, blocks can form a coherent graph, or [Merkle DAG](https://docs.ipfs.io/guides/concepts/merkle-dag/).
+
+Applications interact with IPLD via the Data Model, and IPLD handles marshalling and unmarshalling via a suite of codecs. IPLD codecs may support the complete Data Model or part of the Data Model. Two codecs that support the complete Data Model are [DAG-CBOR](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-cbor.md) and [DAG-JSON](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-json.md). These codecs are respectively based on the CBOR and JSON serialization formats but include formalizations that allow them to encapsulate the IPLD Data Model (including its link type) and additional rules that create a strict mapping between any set of data and its respective content address (or hash digest). These rules include the mandating of particular ordering of keys when encoding maps, or the sizing of integer types when stored.
+
+Filecoin uses the **DAG-CBOR** codec for the serialization and deserialization of its data structures and interacts with that data using the IPLD Data Model and various tools that are built for the IPLD Data Model. Filecoin makes use of IPLD [Selectors]([selectors](https://github.com/ipld/specs/blob/master/selectors/selectors.md)) to address and retrieve subsets of linked data graphs. IPLD [Paths](https://github.com/ipld/specs/blob/master/data-model-layer/paths.md) are also used to address specific nodes within a linked data structure.
+
+IPLD provides a consistent and coherent abstraction above data that allows Filecoin to build and interact with complex, multi-block data structures, such as its HAMT and Sharray <!-- TODO: links, correct names, other multi-block structures, AMT ? -->.
```

Comment:
Needs sanity checking, I'm doing this off an outdated and incomplete understanding of the data structures in Filecoin. They're arguably not strictly an IPLD concern the way they're implemented in Filecoin, but they still rest on top of IPLD and we're doing some work in the multi-block data structure space and have some specs that are _very close_ to Filecoin's HAMT and Sharray. See https://github.com/ipld/specs/tree/master/data-structures - it probably wouldn't take much content to describe precisely how the specs outlined there differ from the Filecoin ones. But maybe there's already good enough docs for them in this repo already.